File: backend/src/main/java/com/prepunite/roadmap/entity/Roadmap.java
package com.prepunite.roadmap.entity;

import com.prepunite.common.entity.BaseEntity;
import com.prepunite.company.entity.Company;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "roadmaps")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Roadmap extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id")
    private Company company;

    @Column(nullable = false, length = 500)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "is_published", nullable = false)
    @Builder.Default
    private Boolean isPublished = true;

    @OneToMany(mappedBy = "roadmap", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("stepOrder ASC")
    @Builder.Default
    private List<RoadmapStep> steps = new ArrayList<>();
}

File: frontend/src/types/roadmap.ts
export interface RoadmapStep {
  id: string;
  stepOrder: number;
  title: string;
  description?: string;
}

export interface Roadmap {
  id: string;
  companyId?: string;
  companyName?: string;
  companySlug?: string;
  title: string;
  description?: string;
  isPublished: boolean;
  steps: RoadmapStep[];
  createdAt: string;
}
